### PR TITLE
Changes done for e2e-tests.

### PR DESCRIPTION
**Pull request by vikasrathee-cs**
Changes done for e2e-tests.

### CODE REVIEW

File: src/e2e-test/features/zendesk/DesignTimeValidation.feature [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+@ZendeskSource
+@Smoke
+@Regression
+Feature: Zendesk Source - Design time validation scenarios
```
Comment (rmstar) [inline src/e2e-test/features/zendesk/DesignTimeValidation.feature]: Please add these tests under a zendesk/source or zendesksource directory

File: src/e2e-test/features/zendesk/RunTime.feature [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    And Connect source as "Zendesk" and sink as "BigQuery" to establish connection
+    And Navigate to the properties page of plugin: "BigQuery"
+    And Enter input plugin property: "referenceName" with value: "Reference"
+    And Configure BigQuery sink plugin for Dataset and Table
+    And Validate "BigQuery" plugin properties
+    And Close the Plugin Properties page
+    And Save and Deploy Pipeline
+    And Run the Pipeline in Runtime
+    And Wait till pipeline is in running state
+    Then Verify the pipeline status is "Succeeded"
```
Comment (rmstar) [inline src/e2e-test/features/zendesk/RunTime.feature]: For tests where we deploy and run a pipeline, please add a step to verify that the correct data was written to the sink.

File: src/e2e-test/features/zendesk/DesignTimeValidation.feature (hunk 1 above)
Comment (Vipinofficial11) [reply src/e2e-test/features/zendesk/DesignTimeValidation.feature]: Added.

File: src/e2e-test/features/zendesk/RunTime.feature (hunk 2 above)
Comment (Vipinofficial11) [reply src/e2e-test/features/zendesk/RunTime.feature]: Added the step, Waiting for Github Actions PR to get merged first.

File: pom.xml [hunk 3]
```diff
@@ -272,6 +275,11 @@
       <version>${mockito.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>ch.qos.logback</groupId>
```
Comment (rmstar) [inline pom.xml]: if this dependency is needed for e2e tests, can we add it under e2e profile?

Comment (Vipinofficial11) [reply pom.xml]: Added the dependency in e2e profile (1c5416086caeac07c30c51f31681c5a5a99aa2c9)

**Review by rmstar — APPROVED** [review]
please squash commits

**Comment by Vipinofficial11** [comment]
> 

Commits Squashed

**Pull request merged**